gnomad_qc/v3/create_release/make_var_annot_hists.py

```diff
@@ -1,61 +1,164 @@
+
+    :param AC: Field in input that contains the allele count information
+    :param AF: Field in input that contains the allele frequency information
+    :return: Expression containing bin name
+    :rtype: hl.expr.StringExpression
+    """
+    bin_expr = (
+        hl.case()
+            .when(AF < 0.0005, "under_0.0005")
+            .when((AF >= 0.0005) & (AF <= 1), "over_0.0005")
+            .default(hl.null(hl.tstr))
+    )
+    return bin_expr
+
+
 def main(args):
     hl.init(default_reference='GRCh38', log='/variant_histograms.log')
 
-    ht = hl.read_table(release_ht_path())
+    logger.info("Loading ANNOTATIONS_HISTS dictionary...")
+    if not file_exists(annotation_hists_path()):
+        raise DataException(
+            "Annotation hists JSON file not found. Need to create this JSON before running script!"
+        )
+
+    with hl.hadoop_open(annotation_hists_path()) as a:
+        ANNOTATIONS_HISTS = json.loads(a.read())
+
+    # NOTE: histogram aggregations on these metrics are done on the entire callset (not just PASS variants), on raw data
+    metrics = list(ANNOTATIONS_HISTS.keys())
+
+    ht = hl.read_table(release_ht_path(public=False))
+
     # NOTE: histogram aggregations are done on the entire callset (not just PASS variants), on raw data
+    ht = ht.select(freq=ht.freq, info=ht.info.select(*metrics))
 
-    hist_dict = ANNOTATIONS_HISTS
-    hist_dict['MQ'] = (20, 60, 40) # Boundaries changed for v3, but could be a good idea to settle on a standard
+    logger.info("Getting info annotation histograms...")
     hist_ranges_expr = get_annotations_hists(
-        ht,
-        ANNOTATIONS_HISTS
+        ht, ANNOTATIONS_HISTS, LOG10_ANNOTATIONS
     )
 
     # NOTE: run the following code in a first pass to determine bounds for metrics
     # Evaluate minimum and maximum values for each metric of interest
     # This doesn't need to be run unless the defaults do not result in nice-looking histograms.
     if args.first_pass:
+        logger.info(
+            "Evaluating minimum and maximum values for each metric of interest, and capping maximum value at 1e10"
+        )
         minmax_dict = {}
-        for metric in hist_ranges_expr.keys():
-            minmax_dict[metric] = hl.struct(min=hl.agg.min(ht[metric]), max=hl.if_else(hl.agg.max(ht[metric])<1e10, hl.agg.max(ht[metric]), 1e10))
+        for metric in metrics:
+            minmax_dict[metric] = hl.struct(
+                min=hl.agg.min(ht.info[metric]),
+                max=hl.if_else(
+                    hl.agg.max(ht.info[metric]) < 1e10,
+                    hl.agg.max(ht.info[metric]),
+                    1e10,
+                ),
+            )
         minmax = ht.aggregate(hl.struct(**minmax_dict))
-        print(minmax)
+        logger.info(f"Metrics bounds: {minmax}")
     else:
-        # Aggregate hists over hand-tooled ranges
+        logger.info(
+            "Aggregating hists over ranges determined using first pass run..."
```

Comment:
the defaults (`annotation_hists_path`) can be used without a first pass, but we usually run the first pass and I also usually make histograms visually too just because I find it helpful. But yes that logger is confusing as it implies only the first pass values will be used not that you actually need to change the values in `annotation_hists_path` after doing the first pass if you want to use them

Reply:
@mike-w-wilson I changed to `determine_bounds`, let me know if you think the changes I made address this comment, or if there is something else that would help